sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -624,20 +624,40 @@ class HDFSFileCatalog(
 
   def getStatus(path: Path): Array[FileStatus] = leafDirToChildrenFiles(path)
 
+  private implicit class LocatedFileStatusIterator(iterator: RemoteIterator[LocatedFileStatus])
```

Comment:
It's used [here][1]. `RemoteIterator` doesn't extend from `Iterator`. This implicit conversion helps to convert contents behind a `RemoteIterator` into an array.

[1]: https://github.com/apache/spark/pull/12153/files#diff-40c347747af9101e7e9fee52fc4120b8R656